app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -92,35 +152,101 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
         });
     }
 
+    private void updateUserStats(int starIncrease, int trophiesIncrease, boolean won) {
+        account.changeStars(starIncrease);
+        account.changeTrophies(trophiesIncrease);
+        account.changeAverageRating(starIncrease);
```

Comment:
Why are you putting starIncrease into averageRating? What do stars have to do with the average rating?

Reply:
The stars you won during the match are the sum of the rankings. And I think we store the total stars in averageRating and divide it by totalMatches when we need to display it (I think)

Reply:
But averageRating espects a number between 0 and 5. In the tests, starIncrease is 21. This gives a precondition error.

Reply:
The preconditions accept a rating between 0 and 25 so this line may be correct

Reply:
Yes if the preconditions have been changed, this line is correct